Source/WebGPU/WGSL/ConstantValue.h

```diff
@@ -36,22 +36,38 @@ namespace WGSL {
 #if HAVE(FP16_HALF_SUPPORT)
 using half = __fp16;
 #else
+// Wrap _Float16, which is the 16bit float type in C++23
 struct half {
+    using f16 = _Float16;
     half()
     {
     }
 
+    half(f16 val)
+        : value(val) { }
+    half(double val)
+        : value(static_cast<f16>(val)) { }
+    half(float val)
+        : value(static_cast<f16>(val)) { }
+    half(long int val)
+        : value(static_cast<f16>(val)) { }
+    half(int val)
+        : value(static_cast<f16>(val)) { }
+    half(unsigned val)
+        : value(static_cast<f16>(val)) { }
+    half(bool val)
+        : value(static_cast<f16>(val)) { }
+
+    // Convert from ConstantResult
     half(auto val)
-        : value(static_cast<float>(val))
-    {
-    }
+        : value(val.value().getHalf().value) { }
 
     operator float() const
     {
         return static_cast<float>(value);
     }
 
-    __fp16 value { 0.f };
```

Comment:
I think we are still compiling with C++20, can you use `__fp16` on `PLATFORM(COCOA)` and `_Float16` on Linux please?

Reply:
or `PLATFORM(__APPLE__)` works too